lib/matplotlib/scale.py

```diff
@@ -478,10 +478,111 @@ def get_transform(self):
         return self._transform
 
 
+def _mask_non_logit(a):
+    """
+    Return a Numpy masked array where all values outside ]0, 1[ are
+    masked. If all values are inside ]0, 1[, the original array is
+    returned.
+    """
+    a = a.copy()
```

Comment:
Ok, so I think keeping the logit _mask and _clip functions analogous to the log ones (_mask_non_positives etc) is good for the codebase, because they have the exact same funcion.

But the *_non_positives functions have bugs, i.e. the docstring is wrong (it mentions masked arrays) and the _clip function modifies user input. So I'll correct the _non_positive function as well, I am against a PR with the logit/log parts of the code doing different things.